sql/parser/regexp_cache.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Nathan VanBenschoten (nvanbenschoten@gmail.com)
+
+package parser
+
+import (
+	"regexp"
+	"sync"
+
+	"github.com/cockroachdb/cockroach/util/cache"
+)
+
+type regexpCacheKey interface {
```

Comment:
I think this deserve a short comment documenting why it exists and what `pattern()` should return.